File: ms_relatorio/src/main/java/com/saude_mais/ms_relatorio/entities/Medico.java
package com.saude_mais.ms_relatorio.entities;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Medico {

    private String id;
    private String nomeMed;
    private String crm;
    private String especialidade;
    private String turnoAtendimento;

    private List<Agendamento> agendamento;
}

File: validationSchema/ms_gerenciamento/medico.js
use clinica;

db.createCollection("medico", {
    validator: {
        $jsonSchema: {
            required: ["nomeMed", "crm", "especialidade", "turnoAtendimento", "agendamento"],
            properties: {
                nomeMed: {
                    bsonType: "string",
                    description: "nome do medico"
                },
                crm: {
                    bsonType: "string",
                    maxLength: 8,
                    description: "crm do medico"
                },
                especialidade: {
                    bsonType: "string",
                    maxLength: 30,
                    description: "especialidade do medico"
                },
                turnoAtendimento: {
                    bsonType: "string",
                    maxLength: 8,
                    description: "turno de atendimento"
                },
            }, action: {
                bsonType: "string", description: "status da consulta",
            }
        }
    }
})

